addons/mass_mailing_sms/models/sms_tracker.py

```diff
@@ -0,0 +1,66 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class SmsTracker(models.Model):
```

Comment:
Ok, there is something bothering me about the new sms.tracker model and the relationships between that, sms.sms, and mailing.trace.

If I'm not mistaken, you currently have this:

```
sms.sms
    mailing_trace_ids (existing)
    uuid (new)

mail.notification
    sms_id (existing)

mailing.trace
    sms_sms_id (existing)

sms.tracker (new)
    sms_uuid (matches uuid from sms)
    mail_notification_id 
    mailing_trace_id
```

First of all, it feels like the "uuid" is tied to the tracking of the SMS, and should be put inside "sms.tracker" and not inside "sms.sms".

The link between a "sms.sms" and a "sms.tracker" would instead be a sms_id field inside the sms.tracker model.
And these trackers would be created by the service actually sending the sms'es to the third party.

Currently, the trackers are being manually created by the various usages, e.g when creating mail.notification or when creating mailing.trace, which feels a bit like it's "not their job".

Instead, the uuid based tracking logic should be independent inside the sms.tracker model (you already kind of did that, so that's good).

And inside the update process, when receiving the statuses for some UUIDs, you would first retrieve the sms.tracker records, then the matching sms'es through the sms_id field, then use those sms records to tie the updates to their related mail.notifications and mailing.traces, as they have a sms_id field.
(All this logic would still be inside sms.tracker).

Meaning we would end up with something like this:
```
sms.sms
    mailing_trace_ids (existing)

mail.notification
    sms_id (existing)

mailing.trace
    sms_sms_id (existing)

sms.tracker (new)
    sms_id
    uuid
```

That way we have less fields and relationships and the functional flows don't have to manually create sms.trackers, those are automatically created and handled by the sms sending process.

The thing we loose though is the cascade deletion.
Which could however be re-introduced with some unlink overrides at notification and trace levels?

And also sms.trackers are always created with this approach, instead of on demand, which may be something we don't want? Not entirely sure on that one.

I know you and Thibault have discussed a lot about this and I'm not the most knowledgeable about mailing/sms flows so this may be wrong for x,y,z reasons, in which case I'm interested to know. I'm not asking you to change anything right away, just opening the door for conversation 🙂

Reply:
PS: Thibault prefers "sms.tracker" so keep that